Tests/SwiftGenTests/ConfigTests.swift

```diff
@@ -0,0 +1,267 @@
+//
+// SwiftGen
+// Copyright (c) 2015 Olivier Halligon
+// MIT Licence
+//
+
+import XCTest
+import PathKit
+
+class ConfigTests: XCTestCase {
+
+  func testReadConfigWithParams() throws {
+    guard let path = Bundle(for: type(of: self)).path(forResource: "config-with-params", ofType: "yml") else {
+      fatalError("Fixture not found")
+    }
+    let file = Path(path)
+    do {
+      let config = try Config(file: file)
+      XCTAssertEqual(config.outputDir, "Common/Generated")
+      XCTAssertEqual(Array(config.commands.keys), ["strings"])
+      guard let stringsEntries = config.commands["strings"] else {
+        return XCTFail("Missing key 'strings'")
+      }
+      XCTAssertEqual(stringsEntries.count, 1)
+      guard let entry = stringsEntries.first else {
+        return XCTFail("Can't get strings only config entry")
+      }
+      XCTAssertEqual(entry.paths, ["Sources1/Folder"])
+      if case TemplateRef.name("structured-swift3") = entry.template { /* OK */ } else {
+        XCTFail("Unexpected template")
+      }
+      XCTAssertEqual(entry.output, "strings.swift")
+      XCTAssertEqual(entry.parameters["foo"] as? Int, 5)
```

Comment:
Anyway, these current tests seem a bit hard to read? Especially for params, where we could just test if the parameters are equal to a dictionary we provide (see code in SwiftGenKit for testing dictionaries).

Reply:
You means [this](https://github.com/SwiftGen/SwiftGenKit/blob/master/Tests/SwiftGenKitTests/TestsHelper.swift#L19-L72)? Yeah, would definitely help readability I guess!